src/itwinai/torch/raytune.py

```diff
@@ -0,0 +1,123 @@
+from ray.tune.schedulers import (
+    AsyncHyperBandScheduler,
+    HyperBandForBOHB,
+    HyperBandScheduler,
+    PopulationBasedTraining,
+)
+from ray.tune.schedulers.pb2 import PB2  # Population Based Bandits
+from ray.tune.search.bayesopt import BayesOptSearch
+from ray.tune.search.bohb import TuneBOHB
+from ray.tune.search.hyperopt import HyperOptSearch
+
+
+def get_raytune_search_alg(tune_config, seeds=False):
+    if "scheduler" in tune_config:
+        scheduler = tune_config["scheduler"]["name"]
+    else:
+        scheduler = None
+    search_alg = tune_config["search_alg"]["name"]
+    metric = tune_config["metric"]
+    mode = tune_config["mode"]
+
+    if (scheduler == "pbt") or (scheduler == "pb2"):
+        if search_alg is not None:
+            print(
+                "INFO: Using schedule '{}' \
+                    is not compatible with Ray Tune search algorithms.".format(scheduler)
+            )
+            print(
+                "INFO: Using the Ray Tune '{}' scheduler without search algorithm".format(
+                    scheduler
+                )
+            )
+        return None
+
+    if (scheduler == "bohb") or (scheduler == "BOHB"):
+        print("INFO: Using TuneBOHB search algorithm since it is required for BOHB shedule")
+        if seeds:
+            seed = 1234
+        else:
+            seed = None
+        return TuneBOHB(
+            metric=metric,
+            mode=mode,
+            seed=seed,
+        )
+
+    # requires pip install bayesian-optimization
```

Comment:
So adding all into one dedicated group in the pyproject.toml? So far I have been installing the one I need directly in the slurm script, because I think that adding a dedicated dependency group for each of them is too much and you might not want to install all of them just to use one, if that makes sense. But I could definitely add them, to be fair they should all be pretty fast to install...

Reply:
Okay I've added them in a dedicated 'hpo' group in the `pyproject.toml`, let me know if you would like me to change it